common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java

```diff
@@ -692,6 +694,82 @@ public AcquireTokenResult acquireDeviceCodeFlowToken(
         return acquireTokenResult;
     }
 
+    @Override
+    public GenerateShrResult generateSignedHttpRequest(
+            @NonNull final GenerateShrCommandParameters parameters) throws Exception {
+        final Context context = parameters.getAndroidApplicationContext();
+        final IClockSkewManager clockSkewManager = new ClockSkewManager(context);
+        final OAuth2TokenCache cache = parameters.getOAuth2TokenCache();
+        final String clientId = parameters.getClientId();
+        final String homeAccountId = parameters.getHomeAccountId();
+        final IPoPAuthenticationSchemeParams popSchemeParams = parameters.getPopParameters();
+
+        if (userHasLocalAccountRecord(cache, clientId, homeAccountId)) {
+            // Perform the signing locally...
+            return generateSignedHttpRequestInternal(context, clockSkewManager, popSchemeParams);
+        } else {
+            throw new UiRequiredException(NO_ACCOUNT_FOUND, "Account does not exist.");
```

Comment:
Make this behave like the broker -- don't throw an `Exception`, instead, return an initialized error

Reply:
0ae490703e69399c53f0d95c2bbd5c1b207a2e76